DA-Database/lib/DA/Database.pm

```diff
@@ -486,6 +486,22 @@ sub _rev_idx_word ($self, $word){
   }
 }
 
+sub get_user_favourites ($self, $id){
+	my @ans;
+	my $sth = $self->dbh->prepare('SELECT word.word, word.sense FROM user INNER JOIN favourite on user.role_id=? INNER JOIN word ON favourite.username=user.username');
```

Comment:
fiz-me entender mal quando te fiz o pedido. A função já recebe o `user_name` por isso basta-te o inner join dos `favourite` com a `word`

Reply:
algo como `select word.word, word.sense from favourite inner join word where favourite.word_id = word.word_id and favourite.username = ?`

Reply:
Pois, por isso é que eu tinha ficado na dúvida, até porque existe diferentes users com o mesmo role_id